src/main/java/seedu/duke/Storage.java

```diff
@@ -0,0 +1,106 @@
+
+    private final File file;
+
+    /**
+     * Initializes storage with the path to the data file.
+     * @param filePath The relative path to the file which data is to be loaded from and saved to.
+     */
+    public Storage(String filePath) {
+        this.file = new File(filePath);
+    }
+
+    /**
+     * Loads the task list data from the file and returns the task list as a List of tasks.
+     * If the file does not exist, create it and then return an empty list as the task list.
+     * @return List of tasks.
+     * @throws DukeException Throws DukeException when the task cannot be loaded from the file.
+     */
+    public List<Task> load() throws DukeException {
+        try {
+            if (this.file.exists()) {
+                ArrayList<Task> taskList = new ArrayList<>();
+                Scanner scn = new Scanner(this.file);
+                while (scn.hasNextLine()) {
+                    String[] data = scn.nextLine().split(SEPARATOR);
+                    switch (data[0]) {
+                    case Storage.TODO_FORMAT:
+                        taskList.add(new Todo(data[2], convertFormatToIsDone(data[1])));
+                        break;
+                    case Storage.EVENT_FORMAT:
+                        taskList.add(new Event(data[2], convertFormatToIsDone(data[1]), data[3]));
+                        break;
+                    case Storage.DEADLINE_FORMAT:
+                        taskList.add(new Deadline(data[2], convertFormatToIsDone(data[1]), data[3]));
+                        break;
+                    }
+                }
+                scn.close();
+                return taskList;
+            } else {
+                this.file.getParentFile().mkdirs();
+                this.file.createNewFile();
+                return new ArrayList<>();
+            }
+        } catch (IOException e) {
+            throw new DukeException("Sorry, there is an error loading the data");
+        }
+    }
+
+    /**
+     * Saves the task list provided to the file.
+     * @param tasks The task list to be saved.
+     * @throws DukeException Throws DukeException when the task list cannot be saved to the file.
+     */
+    public void saveTaskList(TaskList tasks) throws DukeException {
+        FileWriter fileWriter;
+        try {
+            fileWriter = new FileWriter(this.file);
+            for (Task task : tasks.getListOfTasks()) {
+                if (task instanceof Todo) {
+                    fileWriter.write(String.join(Storage.SEPARATOR,
+                            Storage.TODO_FORMAT, convertIsDoneToFormat(task.getIsDone()), task.getDescription()));
+                } else if (task instanceof Event) {
+                    fileWriter.write(String.join(Storage.SEPARATOR,
+                            Storage.EVENT_FORMAT, convertIsDoneToFormat(task.getIsDone()), task.getDescription(),
+                            ((Event) task).getAt()));
+                } else if (task instanceof Deadline) {
```

Comment:
I've read somewhere that instanceof is not ideal for checking class identity -- perhaps there is a better way to approach this?